reframe/core/pipeline.py

```diff
@@ -917,6 +912,55 @@ def __rfm_init__(self, name=None, prefix=None):
         # Disabled hooks
         self._disabled_hooks = set()
 
+    def _append_parameters_to_name(self):
+        if self._rfm_param_space.params:
+            return '_' + '_'.join([util.toalphanum(str(self.__dict__[key]))
+                                   for key in self._rfm_param_space.params])
+        else:
+            return ''
+
+    @classmethod
+    def _process_hook_registry(cls):
+        '''Process and validate the pipeline hooks.'''
+
+        _pipeline_hooks = {}
+        for stage, hooks in cls.pipeline_hooks().items():
+            stage_name = '_'.join(stage.split('_')[1:])
```

Comment:
I don't understand the purpose of `'_'.join(...)` here...

Reply:
I mean, I understand the `split()` part and I agree with that, but I don't understand why we need the `join()`.

Reply:
Ok, got it, I have a better suggestion. I will comment in the updated PR.